addons/sale_mrp/tests/test_sale_mrp_account.py

```diff
@@ -21,3 +21,16 @@ def test_mo_analytic_distribution(self):
         self.sale_order.action_confirm()
         self.assertTrue(self.sale_order.mrp_production_ids.analytic_distribution)
         self.assertTrue(self.sale_order.mrp_production_ids.analytic_account_ids)
+
+    def test_mo_analytic_distribution_from_sol(self):
+        """ ensure analytic account/distribution is inherited from the SO lines
+            when none is set on the bom and without analytic_account_id
+        """
+        analytic_plan = self.env['account.analytic.plan'].create({
+            'name': 'Plan',
+        })
+        self.sale_order.order_line.analytic_distribution = {analytic_plan.id: 100.0}
+        self.sale_order.action_confirm()
+        self.assertEqual(self.sale_order.mrp_production_ids.analytic_distribution, self.sale_order.order_line.analytic_distribution,
+            'The Analytic distribution should be the inherited'
+        )
```

Comment:
```suggestion
        self.assertEqual(self.sale_order.mrp_production_ids.analytic_distribution, self.sale_order.order_line.analytic_distribution)
```
The assert is clear enough